src/app-layout/drawer/index.tsx

```diff
@@ -59,12 +47,75 @@ export function Drawer({
   onToggle,
   onClick,
   onLoseFocus,
+  drawers,
+  onResize,
+  size,
+  getMaxWidth,
+  refs,
 }: DesktopDrawerProps) {
+  const openButtonWrapperRef = useRef<HTMLElement | null>(null);
   const { TagName, iconName, getLabels } = togglesConfig[type];
   const { mainLabel, closeLabel, openLabel } = getLabels(ariaLabels);
   const drawerContentWidthOpen = isMobile ? undefined : width;
   const drawerContentWidth = isOpen ? drawerContentWidthOpen : undefined;
-  const openButtonWrapperRef = useRef<HTMLElement | null>(null);
+
+  const activeDrawer = drawers?.items.find(item => item.id === drawers.activeDrawerId);
+
+  const MIN_WIDTH = activeDrawer?.size && activeDrawer.size < 280 ? activeDrawer?.size : 280;
+  const [relativeSize, setRelativeSize] = useState(0);
+  const focusVisible = useFocusVisible();
+
+  useEffect(() => {
+    // effects are called inside out in the components tree
+    // wait one frame to allow app-layout to complete its calculations
+    const handle = requestAnimationFrame(() => {
+      const maxSize = getMaxWidth();
+      setRelativeSize((size / maxSize) * 100);
+    });
+    return () => cancelAnimationFrame(handle);
+  }, [size, getMaxWidth]);
+
+  const setSidePanelWidth = (width: number) => {
+    const maxWidth = getMaxWidth();
+    const size = getLimitedValue(MIN_WIDTH, width, maxWidth);
+
+    if (isOpen && maxWidth >= MIN_WIDTH) {
+      onResize({ size });
+    }
+  };
+
+  const position = 'side';
+  const setBottomPanelHeight = () => {};
```

Comment:
Yeah I should have left a comment here. I am currently importing directly from the Split Panel to avoid copy/ pasting code, and didn't want to touch any of that code to avoid unexpected regressions. I think ideally, we extract the resize functions into a neutral folder, and have Split Panel and Drawer both use them. For now, I was going to leave like this. The other option is to copy the functions into the drawer code and refactor from there.

Reply:
ok, let's leave it as is for now (except extracting into `ResizableDrawer`)